arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/Stream.kt

```diff
@@ -1322,6 +1323,69 @@ import kotlin.random.Random
   fun delayBy(d: Duration): Stream<O> =
     sleep_(d).append { this }
 
+  /**
+   * Runs the supplied stream in the background as elements from this stream are pulled.
+   *
+   * The resulting stream terminates upon termination of this stream. The background stream will
+   * be interrupted at that point. Early termination of [other] does not terminate the resulting stream.
+   *
+   * Any errors that occur in either `this` or [other] stream result in the overall stream terminating
+   * with an error.
+   *
+   * Upon finalization, the resulting stream will interrupt the background stream and wait for it to be
+   * finalized.
```

Comment:
No, there is no difference from `parJoin`.

```
val one = Stream(1)
val other = Stream.bracket({ Unit }, { sleep(1.second) }).flatMap { never<Unit>() }

one.concurrently(other)
```

`one` finishes immediately, but it waits until `other` is finished finalizing.
So here it will wait until the `sleep(1.seconds)` is finished, before continuing to for example an `.append { }`.

Reply:
👍 so what'd be the rationale for offering both?

Reply:
In `parJoin` all inner streams run as child processes of the outer parent process.
With `concurrent` `this` is parent stream.

Reply:
Ok, then there is a relevant difference 👍